core/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -75,7 +76,22 @@ public final void handleRequest(RestRequest request, RestChannel channel, NodeCl
         }
 
         // execute the action
-        action.run();
+        action.accept(channel);
+    }
+
+    @FunctionalInterface
```

Comment:
It doesn't make the linter mad, so I pushed a1f065e8e1d3a09b52c3f84e5230630b24ae3056.